### PR TITLE
Predicate processing

### PR DESCRIPTION
**Pull request by mmatera**
This is the part of ConditionalExpressions that involves advanced logical processing.

### CODE REVIEW

File: mathics/builtin/inference.py [hunk 1]
```diff
@@ -0,0 +1,381 @@
...
+        return None
+
+    # check for consistency:
+    consistent_assumptions = Expression("And", *assumptions_list)
+    val_consistent_assumptions = consistent_assumptions.evaluate(evaluation)
+    if val_consistent_assumptions == SymbolFalse:
+        evaluation.message("Inconsistent assumptions")
+
+    if assumptions_list is None:
+        return remove_nots_when_unnecesary(pred, evaluation).evaluate(evaluation)
```
Comment (rocky) [inline mathics/builtin/inference.py:274]: pred is not defined here.

File: mathics/builtin/inference.py [hunk 2]
```diff
@@ -0,0 +1,381 @@
...
+                            SymbolFalse,
+                        )
+                    )
+            else:
+                assumption_rules.append(Rule(pat, SymbolFalse))
+        else:
+            assumption_rules.append(Rule(pat, symbol_value))
+    # TODO: expand the pred and assumptions into an standard,
+    # atomized form, and then apply the rules...
+    print("assumptions:")
```
Comment (rocky) [inline mathics/builtin/inference.py:346]: Is this print and the one below supposed to be here? 

Change this in the rebase version.

**Comment by rocky** [comment]
@mmatera the code here doesn't look like it is ready for a PR. What's up?

**Comment by mmatera** [comment]
Superseded by #1337

**Comment by mmatera** [comment]
Superseded by #1337

**Pull request closed without merging**